frontend/awx/resources/notifications/hooks/useNotificationColumns.tsx

```diff
@@ -0,0 +1,42 @@
+import { useCallback, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ITableColumn, usePageNavigate } from '../../../../../framework';
+import { AwxRoute } from '../../../main/AwxRoutes';
+import { useNameColumn } from '../../../../common/columns';
+import { NotificationTemplate } from '../../../interfaces/NotificationTemplate';
+
+export function useNotificationsColumns(options?: {
+  disableSort?: boolean;
+  disableLinks?: boolean;
+}) {
+  const { t } = useTranslation();
+  const pageNavigate = usePageNavigate();
+  const nameClick = useCallback(
+    (notificationTemplate: NotificationTemplate) => {
+      return pageNavigate(AwxRoute.NotificationTemplatePage, {
+        params: {
+          id: notificationTemplate.id,
+        },
+      });
+    },
+    [pageNavigate]
+  );
+  const nameColumn = useNameColumn({ ...options, onClick: nameClick });
+  const typeColumn = useMemo<ITableColumn<NotificationTemplate>>(
+    () => ({
+      header: t('Type'),
+      type: 'text',
+      value: (notificationTemplate: NotificationTemplate) => {
+        return notificationTemplate.notification_type;
+      },
+      card: 'subtitle',
+      list: 'subtitle',
+    }),
+    [t]
+  );
+  const tableColumns = useMemo<ITableColumn<NotificationTemplate>[]>(
```

Comment:
Do we want a header of Options above the start, success, failure toggles?

Reply:
I'm not entirely sure. The old UI does have them but I wasn't able to figure out how to use the switch toggle within a column. It seems that the new UI uses the switch toggle component in the action part of the row. I could be wrong of course :sweat_smile: